tests/widget_test.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+import pytest
+from ess.reduce.parameter import Parameter
+from ess.reduce.widgets import SwitchWidget, create_parameter_widget
+
+
+@pytest.mark.filterwarnings(
+    'ignore::DeprecationWarning'
+)  # Ignore deprecation warning from widget library
+def test_switchable_widget_dispatch() -> None:
+    switchable_param = Parameter('a', 'a', 1, switchable=True)
+    assert isinstance(create_parameter_widget(switchable_param), SwitchWidget)
+    non_switchable_param = Parameter('b', 'b', 2, switchable=False)
+    assert not isinstance(create_parameter_widget(non_switchable_param), SwitchWidget)
+
+
+@pytest.mark.filterwarnings(
+    'ignore::DeprecationWarning'
+)  # Ignore deprecation warning from widget library
+def test_collect_values_from_disabled_switchable_widget() -> None:
+    from ess.reduce.ui import collect_values
```

Comment:
I think you are kind of testing an implementation detail here: Someone might remove the `collect_values` function, change the mechanism, etc. When that happens, this test will need to be refactored, or worse, it will be removed.

Have you considered testing the actual observable behavior we want, i.e., create a `Pipeline` and check if those values are set by the UI? Is this too complicated to do in the current design? Would something like MVP or MVC make this easier?

Reply:
Yes I would rather do that once we have #68  done...?